include/fastdds/dds/topic/DataWriterListener.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file DataWriterListener.hpp
+ */
+
+#ifndef _FASTRTPS_DATAWRITERLISTENER_H_
+#define _FASTRTPS_DATAWRITERLISTENER_H_
+
+#include <fastdds/rtps/common/Types.h>
+#include <fastdds/rtps/common/MatchingInfo.h>
+#include <fastrtps/qos/DeadlineMissedStatus.h>
+#include <fastrtps/qos/LivelinessLostStatus.h>
+
+namespace eprosima {
+namespace fastdds {
+namespace dds {
+
+class DataWriter;
+
+/**
+ * Class DataWriterListener, allows the end user to implement callbacks triggered by certain events.
+ * @ingroup FASTDDS_MODULE
+ */
+class RTPS_DllAPI DataWriterListener
+{
+public:
+    DataWriterListener(){}
+    virtual ~DataWriterListener(){}
+
+    /**
+     * This method is called when the Publisher is matched (or unmatched) against an endpoint.
+     * @param writer Pointer to the associated Publisher
+     * @param info Information regarding the matched subscriber
+     */
+    virtual void on_publication_matched(
```

Comment:
Comment this fucntion and write a TODO. Do the same with all not implemented functions